src/main/java/au/gov/digitalhealth/terminology/amtflatfile/Amt2FlatFile.java

```diff
@@ -183,6 +197,60 @@ public void execute() throws MojoExecutionException, MojoFailureException {
 		}
 	}
 
+    private void validateOutputPath(String outputPath, String expectedMimeType) {
+        try {
+            Path path = Paths.get(outputPath);
+
+            if (Files.isSymbolicLink(path)) {
+                throw new SecurityException("The input ZIP file must not be a symlink for security reasons");
+            }
+
+            if (Files.exists(path, LinkOption.NOFOLLOW_LINKS)) {
+                BasicFileAttributes attr = Files.readAttributes(
+                    path, BasicFileAttributes.class, LinkOption.NOFOLLOW_LINKS);
+
+                if (!attr.isRegularFile()) {
+                    throw new SecurityException(
+                        "The specified output file " + outputFilePath + " exists, but is not a regular file. Cannot be overwritten.");
+                } else if (!tika.detect(path).equals(expectedMimeType)) {
+                    throw new SecurityException(
+                        "The specified output file " + outputFilePath + " exists, but is not a " + expectedMimeType
+                                + " file as expected, detected type was " + tika.detect(path) + ". Cannot be overwritten");
+                }
+            }
+        } catch (IOException e) {
+            throw new IllegalArgumentException("Could not validate input ZIP file path", e);
```

Comment:
Should this be "Could not validate **output** ZIP file path"?

Reply:
yes that's correct, I'll ammend